01-syntax/06-project-elife/elife.js

```diff
@@ -1 +1,399 @@
+    if (!handled) {
+      arguments[0].energy -= 0.2;
+      if (critter.energy <= 0) {
+        this.grid.set(vector, null);
+      }
+    }
+  };
+
+  actionTypes.grow = function () {
+    arguments[0].energy += 0.5;
+
+    return true;
+  };
+
+  actionTypes.move = function (critter, vector, action) {
+    var destination = this.checkDestination(action, vector);
+
+    if (destination === null || critter.energy <= 1 ||
+      this.grid.get(destination) !== null) {
+      return false;
+    }
+
+    arguments[0].energy -= 1;
+    this.grid.set(vector, null);
+    this.grid.set(destination, critter);
+
+    return true;
+  };
+
+  actionTypes.eat = function (critter, vector, action) {
+    var destination = this.checkDestination(action, vector);
+    var atDestination;
+    atDestination = (destination !== null && this.grid.get(destination));
+
+    if (!atDestination || atDestination.energy === null) {
+      return false;
+    }
+
+    arguments[0].energy += atDestination.energy;
+    this.grid.set(destination, null);
+
+    return true;
+  };
+
+  actionTypes.reproduce = function (critter, vector, action) {
+    var baby = elementFromChar(this.legend, critter.originChar);
+    var destination = this.checkDestination(action, vector);
+
+    if (destination === null || critter.energy <= 2 * baby.energy ||
+      this.grid.get(destination) !== null) {
+      return false;
+    }
+
+    arguments[0].energy -= 2 * baby.energy;
+    this.grid.set(destination, baby);
+    return true;
+  };
+
+  function Plant() {
+    this.energy = 3 + Math.random() * 4;
+  }
+
+  Plant.prototype.act = function (view) {
+    var space;
+
+    if (this.energy > 15) {
+      space = view.find(' ');
+      if (space) {
+        return { type: 'reproduce', direction: space };
+      }
+    }
+
+    if (this.energy < 20) {
+      return { type: 'grow' };
+    }
+
+    return undefined;
+  };
+
+  function PlantEater() {
+    this.energy = 20;
+  }
+
+  PlantEater.prototype.act = function (view) {
+    var space = view.find(' ');
+    var plant = view.find('*');
+
+    if (this.energy > 60 && space) {
+      return { type: 'reproduce', direction: space };
+    }
+
+    if (plant) {
+      return { type: 'eat', direction: plant };
+    }
+
+    if (space) {
+      return { type: 'move', direction: space };
+    }
+
+    return undefined;
+  };
+
+  function CritterEater() {
+    this.energy = 30;
+  }
+
+  CritterEater.prototype.act = function (view) {
+    var space = view.find(' ');
+    var critter = view.find('o');
+    var plant = view.find('*');
+
+    if (this.energy > 60 && space) {
+      return { type: 'reproduce', direction: space };
+    }
+
+    if (critter) {
+      return { type: 'eat', direction: critter };
+    }
+
+    if (plant) {
+      return { type: 'eat', direction: plant };
+    }
+
+    if (space) {
+      return { type: 'move', direction: space };
+    }
+
+    return undefined;
+  };
+
+  window.valley = new LifelikeWorld(plan, {
```

Comment:
I propose to rid off `valley` completely and initialize   `LifelikeWorld ` directly in 

https://github.com/nervebassmaster/sf-eng-init-gl-spring-2016/blob/elifeProject/01-syntax/06-project-elife/index.html#L8